### DISCUSSION TITLE
Consider mirroring your repos to Codeberg

### DISCUSSION THREAD

**Discussion by throwaway-d** [ThioJoe/YT-Spammer-Purge#994] [Ideas]
Hello @ThioJoe

**I believe you should mirror your repos to Codeberg** because:
1. Codeberg is libre, free, open-source, privacy-friendly, non-profit
2. GitHub is privacy-invasive, proprietary
3. People on Codeberg like me can contribute to the project
4. GitHub can take down your repos

Here are some sources for you:
1. https://sfconservancy.org/GiveUpGitHub
2. https://docs.codeberg.org/advanced/migrating-repos
3. https://github.com/humanetech-community/awesome-humane-tech/issues/33
4. https://github.com/ungoogled-software/ungoogled-chromium/discussions/1894
5. https://cadence.moe/blog/2022-07-03-git-forge-opinions-github-gitlab-gitea-sourcehut
6. https://github.com/andersju/webbkoll/issues/35

### FAQ
1. **Q:** - GitHub Actions -- this is a huge time saver for me
          - Forcing all contributors to modify their current setup and move to another hosting -- we may just lose contributors in the end
    
   **A:** As I said, we only just _mirror_. The repo will be available and contributable on **both** sites.

2. **Q:** - Transferring issues -- is this possible? Issues here represent a technical knowledge base we can't live without
          - Transferring wiki and all edit history -- possible?

    **A:** Yes, it is possible.
a. Example for issues:

- https://github.com/andersju/webbkoll/issues/35
- https://codeberg.org/dataskydd.net/webbkoll/issues/35
- https://github.com/andersju/webbkoll/issues/33
- https://codeberg.org/dataskydd.net/webbkoll/issues/33
- https://github.com/zyachel/quetre/issues/7
- https://codeberg.org/zyachel/quetre/issues/7

b. Example for commits:
- https://codeberg.org/dataskydd.net/webbkoll/commit/6ddfbc6c71ab92f754252dd4904cf49fb19672a6
- https://github.com/andersju/webbkoll/commit/6ddfbc6c71ab92f754252dd4904cf49fb19672a6

c. Example for wiki:
- https://github.com/Freeyourgadget/Gadgetbridge/wiki/Home/_history?page=2
- https://codeberg.org/Freeyourgadget/Gadgetbridge/wiki/Home?action=_revision

3. **Q:** Despite having mirror/s, the project will still continue to be operated from GitHub for development, so having yet another mirror doesn't solve anything.

   **A:** You can create issues, contribute, view the repos on there without going to GitHub! Example:
- https://github.com/libredirect/libredirect
- https://codeberg.org/LibRedirect/libredirect
- https://github.com/zyachel/quetre
- https://codeberg.org/zyachel/quetre

4. **Q:** How does Codeberg mirror comments from GitHub?
   
   **A:** It uses Access Token. 
   ![image](https://user-images.githubusercontent.com/121148372/209098935-5b16df71-9832-4e2f-8a12-e8911d5e7e66.png)

5. **Q:** My repos are very bandwidth-intensive projects, and I have my doubts as to whether the owner(s) of Codeberg would be prepared for such a massive bandwidth rate increase. I've also had individual projects be disabled on small sites before due to taking too much bandwidth.

   **A:** Oh.. I think you should directly ask https://docs.codeberg.org/contact/ about if they are ready for massive bandwidth rate increase.

**Comment by TechStudent10** [comment]
????????? what does this have to do with the project?

**Reply by throwaway-d** [reply]
This project is on GitHub.

**Comment by throwaway-d** [comment]
BTW, I created a repo dedicated to mirroring to Codeberg: https://codeberg.org/HexagonCDN/Mirror_to_Codeberg
